`npm run build` fails with numerous `error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'

### Summary

On a fresh clone of this repo, following the instructions for a DE org, I get the following on step 2 when running `npm run build`:

```
$ npm run build

> react-recipes@1.59.0 build
> tsc -b && vite build

src/recipes/error-handling/ErrorBoundaryRecipe.test.tsx:68:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

68     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/GraphqlErrors.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:143:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

143     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:151:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

151     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:159:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

159     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/hello/BindingAccountName.test.tsx:55:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

55     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ChildToParent.test.tsx:76:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

76     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ConditionalStatus.test.tsx:69:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

69     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/HelloWorld.test.tsx:28:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

28     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/LifecycleFetch.test.tsx:100:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

100     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ListOfAccounts.test.tsx:78:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

78     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ParentToChild.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/StateManagement.test.tsx:90:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

90     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/integration/DashboardAliasedQueries.test.tsx:89:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

89     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/integration/SearchableAccountList.test.tsx:168:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

168     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/integration/SearchableAccountList.test.tsx:179:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

179     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/CreateRecord.test.tsx:107:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

107     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/DeleteRecord.test.tsx:168:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

168     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/QueryMutationTogether.test.tsx:181:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

181     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/ServerErrorHandling.test.tsx:102:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

102     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/UpdateRecord.test.tsx:160:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

160     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/AliasedMultiObjectQuery.test.tsx:60:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

60     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/FilteredList.test.tsx:121:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

121     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/ImperativeRefetch.test.tsx:86:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

86     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/ListOfRecords.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/PaginatedList.test.tsx:165:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

165     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/RelatedRecords.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/SingleRecord.test.tsx:75:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

75     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/SortedResults.test.tsx:98:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

98     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/LinkDemo.test.tsx:49:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

49     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/NavLinkDemo.test.tsx:52:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

52     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/NestedRoutes.test.tsx:105:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

105     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/routing/RouteParameters.test.tsx:116:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

116     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/routing/UseNavigate.test.tsx:101:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

101     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/ApexRest.test.tsx:138:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

138     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/ConnectApi.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/DisplayCurrentUser.test.tsx:53:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

53     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/UiApiRest.test.tsx:148:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

148     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardDSR.test.tsx:80:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

80     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardShadcn.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardSLDS.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonDSR.test.tsx:60:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

60     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonShadcn.test.tsx:46:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

46     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonSLDS.test.tsx:21:21 - error TS2554: Expected 1 arguments, but got 2.

21       expect(match, `expected a button with class ${cls}`).toBeTruthy();
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonSLDS.test.tsx:55:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

55     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsDSR.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsLucide.test.tsx:41:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

41     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsSLDS.test.tsx:50:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

50     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~


Found 49 errors.
```

Every previous step has succeeded.

According to [this](https://github.com/chaance/vitest-axe/issues/12#issuecomment-2287018197), it's because `vitest.config.ts` includes `globals: true`, and the workaround is to explicitly import `expect` from `vitest` in each `*.test.tsx` file. I've confirmed that workaround is effective locally, but I'm not sure if that's what you want to do here or not. I confess that I'm far from an expert in this area, so there may be better options for resolving this.

### Salesforce Org Type

Scratch Org

### Steps To Reproduce

_No response_

### Current Behavior

_No response_

### Expected Behavior

_No response_

### Relevant Log Output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

Here's an exhaustive session log of the entire process leading up to the failures including killing any running `node` processes before starting the process to ensure it's not picking up something stale:

```
~/projects-> node --version
v22.13.0

~/projects-> npm --version
11.14.0

~/projects-> pskill node
PsKill v1.16 - Terminates processes on local or remote systems
Copyright (C) 1999-2016  Mark Russinovich
Sysinternals - www.sysinternals.com

5 processes named node killed.

~/projects-> pskill node
PsKill v1.16 - Terminates processes on local or remote systems
Copyright (C) 1999-2016  Mark Russinovich
Sysinternals - www.sysinternals.com

Unable to kill process node:
Process does not exist.

~/projects-> git clone https://github.com/trailheadapps/multiframework-recipes.git
Cloning into 'multiframework-recipes'...
remote: Enumerating objects: 390, done.
remote: Counting objects: 100% (390/390), done.
remote: Compressing objects: 100% (269/269), done.
remote: Total 390 (delta 115), reused 330 (delta 88), pack-reused 0 (from 0)
Receiving objects: 100% (390/390), 1.10 MiB | 10.83 MiB/s, done.
Resolving deltas: 100% (115/115), done.

~/projects-> cd multiframework-recipes/

~/projects/multiframework-recipes[main]-> sf org create scratch -d -f config/project-scratch-def.json -a recipes
 ────────────── Creating Scratch Org ──────────────

 ✔ Prepare Request 52ms
 ✔ Send Request 36.06s
 ◯ Wait For Org - Skipped
 ✔ Available 2ms
 ✔ Authenticate 1.85s
 ✔ Deploy Settings 7.49s
 ✔ Done 0ms

 Request Id: 2SRHo00000UcHTaOAN (​https://<dev-hub-host-name>.my.salesforce.com/2SRHo00000UcHTaOAN​)
 OrgId: 00DDE000005hGxd
 Username: test-ryvev4jchbsz@example.com
 Alias: recipes
 Elapsed Time: 45.46s

Your scratch org is ready.
Warning: Record types defined in the scratch org definition file will stop being capitalized by default in a future release.
Set the `org-capitalize-record-types` config var to `true` to enforce capitalization.

~/projects/multiframework-recipes[main]-> sf project deploy start -d force-app/main/default/objects -d force-app/main/default/classes -d force-app/main/default/permissionsets -d force-app/main/default/cspTrustedSites
 ─────────────── Deploying Metadata ───────────────

 Deploying v66.0 metadata to test-ryvev4jchbsz@example.com using the v66.0 SOAP API.

 ✔ Preparing 129ms
 ◯ Waiting for the org to respond - Skipped
 ✔ Deploying Metadata 5.48s
   ▸ Components: 7/7 (100%)
 ◯ Running Tests - Skipped
 ✔ Updating Source Tracking 15ms
   ▸ Members: 7/7 (100%)
 ✔ Done 0ms

 Status: Succeeded
 Deploy ID: 0AfDE00003m8ZA20AM
 Target Org: test-ryvev4jchbsz@example.com
 Elapsed Time: 5.63s


Deployed Source
┌─────────┬──────────────────────┬────────────────┬──────────────────────────────────────────────────────────────────────────────┐
│ State   │ Name                 │ Type           │ Path                                                                         │
├─────────┼──────────────────────┼────────────────┼──────────────────────────────────────────────────────────────────────────────┤
│ Created │ ContactResource      │ ApexClass      │ force-app\main\default\classes\ContactResource.cls                           │
│ Created │ ContactResource      │ ApexClass      │ force-app\main\default\classes\ContactResource.cls-meta.xml                  │
│ Created │ ContactResourceTest  │ ApexClass      │ force-app\main\default\classes\ContactResourceTest.cls                       │
│ Created │ ContactResourceTest  │ ApexClass      │ force-app\main\default\classes\ContactResourceTest.cls-meta.xml              │
│ Created │ ContactsResource     │ ApexClass      │ force-app\main\default\classes\ContactsResource.cls                          │
│ Created │ ContactsResource     │ ApexClass      │ force-app\main\default\classes\ContactsResource.cls-meta.xml                 │
│ Created │ ContactsResourceTest │ ApexClass      │ force-app\main\default\classes\ContactsResourceTest.cls                      │
│ Created │ ContactsResourceTest │ ApexClass      │ force-app\main\default\classes\ContactsResourceTest.cls-meta.xml             │
│ Created │ AmazonS3Media        │ CspTrustedSite │ force-app\main\default\cspTrustedSites\AmazonS3Media.cspTrustedSite-meta.xml │
│ Created │ Contact.Picture__c   │ CustomField    │ force-app\main\default\objects\Contact\fields\Picture__c.field-meta.xml      │
│ Created │ recipes              │ PermissionSet  │ force-app\main\default\permissionsets\recipes.permissionset-meta.xml         │
└─────────┴──────────────────────┴────────────────┴──────────────────────────────────────────────────────────────────────────────┘


~/projects/multiframework-recipes[main]-> sf org assign permset -n recipes
Permsets Assigned
┌───────────────────────────────┬───────────────────────────┐
│ Username                      │ Permission Set Assignment │
├───────────────────────────────┼───────────────────────────┤
│ test-ryvev4jchbsz@example.com │ recipes                   │
└───────────────────────────────┴───────────────────────────┘

~/projects/multiframework-recipes[main]-> sf data tree import -p ./data/data-plan.json
Warning: The plan contains the 'saveRefs' and/or 'resolveRefs' properties. These properties will be ignored and can be removed.
Import Results
┌─────────────────┬─────────────┬────────────────────┐
│ Reference ID    │ Type        │ ID                 │
├─────────────────┼─────────────┼────────────────────┤
│ AccountRef1     │ Account     │ 001DE00003WfRJcYAN │
│ AccountRef2     │ Account     │ 001DE00003WfRJdYAN │
│ AccountRef3     │ Account     │ 001DE00003WfRJeYAN │
│ AccountRef4     │ Account     │ 001DE00003WfRJfYAN │
│ AccountRef5     │ Account     │ 001DE00003WfRJgYAN │
│ AccountRef6     │ Account     │ 001DE00003WfRJhYAN │
│ AccountRef7     │ Account     │ 001DE00003WfRJiYAN │
│ AccountRef8     │ Account     │ 001DE00003WfRJjYAN │
│ AccountRef9     │ Account     │ 001DE00003WfRJkYAN │
│ AccountRef10    │ Account     │ 001DE00003WfRJlYAN │
│ ContactRef1     │ Contact     │ 003DE00002lrZl2YAE │
│ ContactRef2     │ Contact     │ 003DE00002lrZl3YAE │
│ ContactRef3     │ Contact     │ 003DE00002lrZl4YAE │
│ ContactRef4     │ Contact     │ 003DE00002lrZl5YAE │
│ ContactRef5     │ Contact     │ 003DE00002lrZl6YAE │
│ ContactRef6     │ Contact     │ 003DE00002lrZl7YAE │
│ OpportunityRef1 │ Opportunity │ 006DE000013EwtJYAS │
│ OpportunityRef2 │ Opportunity │ 006DE000013EwtKYAS │
│ OpportunityRef3 │ Opportunity │ 006DE000013EwtLYAS │
│ OpportunityRef4 │ Opportunity │ 006DE000013EwtMYAS │
│ OpportunityRef5 │ Opportunity │ 006DE000013EwtNYAS │
│ OpportunityRef6 │ Opportunity │ 006DE000013EwtOYAS │
│ OpportunityRef7 │ Opportunity │ 006DE000013EwtPYAS │
│ OpportunityRef8 │ Opportunity │ 006DE000013EwtQYAS │
└─────────────────┴─────────────┴────────────────────┘

~/projects/multiframework-recipes[main]-> cd force-app/main/react-recipes/uiBundles/reactRecipes/

~/projects/multiframework-recipes/force-app/main/react-recipes/uiBundles/reactRecipes[main]-> npm install
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-modal@3.14.3
npm warn Found: react@19.2.5
npm warn node_modules/react
npm warn   react@"^19.2.0" from the root project
npm warn   74 more (@floating-ui/react-dom, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^0.14.0 || ^15.0.0 || ^16 || ^17" from react-modal@3.14.3
npm warn node_modules/@salesforce/design-system-react/node_modules/react-modal
npm warn   react-modal@"3.14.3" from @salesforce/design-system-react@0.10.65
npm warn   node_modules/@salesforce/design-system-react
npm warn
npm warn Conflicting peer dependency: react@17.0.2
npm warn node_modules/react
npm warn   peer react@"^0.14.0 || ^15.0.0 || ^16 || ^17" from react-modal@3.14.3
npm warn   node_modules/@salesforce/design-system-react/node_modules/react-modal
npm warn     react-modal@"3.14.3" from @salesforce/design-system-react@0.10.65
npm warn     node_modules/@salesforce/design-system-react
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-modal@3.14.3
npm warn Found: react-dom@19.2.5
npm warn node_modules/react-dom
npm warn   react-dom@"^19.2.0" from the root project
npm warn   48 more (@floating-ui/react-dom, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react-dom@"^0.14.0 || ^15.0.0 || ^16 || ^17" from react-modal@3.14.3
npm warn node_modules/@salesforce/design-system-react/node_modules/react-modal
npm warn   react-modal@"3.14.3" from @salesforce/design-system-react@0.10.65
npm warn   node_modules/@salesforce/design-system-react
npm warn
npm warn Conflicting peer dependency: react-dom@17.0.2
npm warn node_modules/react-dom
npm warn   peer react-dom@"^0.14.0 || ^15.0.0 || ^16 || ^17" from react-modal@3.14.3
npm warn   node_modules/@salesforce/design-system-react/node_modules/react-modal
npm warn     react-modal@"3.14.3" from @salesforce/design-system-react@0.10.65
npm warn     node_modules/@salesforce/design-system-react
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-onclickoutside@6.13.2
npm warn Found: react@19.2.5
npm warn node_modules/react
npm warn   react@"^19.2.0" from the root project
npm warn   74 more (@floating-ui/react-dom, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^15.5.x || ^16.x || ^17.x || ^18.x" from react-onclickoutside@6.13.2
npm warn node_modules/@salesforce/design-system-react/node_modules/react-onclickoutside
npm warn   react-onclickoutside@"^6.10.0" from @salesforce/design-system-react@0.10.65
npm warn   node_modules/@salesforce/design-system-react
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^15.5.x || ^16.x || ^17.x || ^18.x" from react-onclickoutside@6.13.2
npm warn   node_modules/@salesforce/design-system-react/node_modules/react-onclickoutside
npm warn     react-onclickoutside@"^6.10.0" from @salesforce/design-system-react@0.10.65
npm warn     node_modules/@salesforce/design-system-react
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-onclickoutside@6.13.2
npm warn Found: react-dom@19.2.5
npm warn node_modules/react-dom
npm warn   react-dom@"^19.2.0" from the root project
npm warn   48 more (@floating-ui/react-dom, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react-dom@"^15.5.x || ^16.x || ^17.x || ^18.x" from react-onclickoutside@6.13.2
npm warn node_modules/@salesforce/design-system-react/node_modules/react-onclickoutside
npm warn   react-onclickoutside@"^6.10.0" from @salesforce/design-system-react@0.10.65
npm warn   node_modules/@salesforce/design-system-react
npm warn
npm warn Conflicting peer dependency: react-dom@18.3.1
npm warn node_modules/react-dom
npm warn   peer react-dom@"^15.5.x || ^16.x || ^17.x || ^18.x" from react-onclickoutside@6.13.2
npm warn   node_modules/@salesforce/design-system-react/node_modules/react-onclickoutside
npm warn     react-onclickoutside@"^6.10.0" from @salesforce/design-system-react@0.10.65
npm warn     node_modules/@salesforce/design-system-react
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-text-truncate@0.19.0
npm warn Found: react-dom@19.2.5
npm warn node_modules/react-dom
npm warn   react-dom@"^19.2.0" from the root project
npm warn   48 more (@floating-ui/react-dom, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react-dom@"^15.4.1 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-text-truncate@0.19.0
npm warn node_modules/@salesforce/design-system-react/node_modules/react-text-truncate
npm warn   react-text-truncate@"^0.19.0" from @salesforce/design-system-react@0.10.65
npm warn   node_modules/@salesforce/design-system-react
npm warn
npm warn Conflicting peer dependency: react-dom@18.3.1
npm warn node_modules/react-dom
npm warn   peer react-dom@"^15.4.1 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-text-truncate@0.19.0
npm warn   node_modules/@salesforce/design-system-react/node_modules/react-text-truncate
npm warn     react-text-truncate@"^0.19.0" from @salesforce/design-system-react@0.10.65
npm warn     node_modules/@salesforce/design-system-react
npm warn deprecated lodash.isequal@4.5.0: This package is deprecated. Use require('node:util').isDeepStrictEqual instead.
npm warn deprecated whatwg-encoding@3.1.1: Use @exodus/bytes instead for a more spec-conformant and faster implementation
npm warn deprecated node-domexception@1.0.0: Use your platform's native DOMException instead
npm warn deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1

added 1206 packages, and audited 1207 packages in 27s

323 packages are looking for funding
  run `npm fund` for details

6 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

~/projects/multiframework-recipes/force-app/main/react-recipes/uiBundles/reactRecipes[main]-> npm run graphql:schema
> react-recipes@1.59.0 graphql:schema
> node scripts/get-graphql-schema.mjs

(node:17848) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Executing introspection query against https://<scratch-org-host-name>.scratch.my.salesforce.com/services/data/v66.0/graphql
Schema saved to C:\Users\Scott\projects\multiframework-recipes\schema.graphql

~/projects/multiframework-recipes/force-app/main/react-recipes/uiBundles/reactRecipes[main]-> npm run graphql:codegen
> react-recipes@1.59.0 graphql:codegen
> graphql-codegen

✔ Parse Configuration
✔ Generate outputs

~/projects/multiframework-recipes/force-app/main/react-recipes/uiBundles/reactRecipes[main]-> npm run build
> react-recipes@1.59.0 build
> tsc -b && vite build

src/recipes/error-handling/ErrorBoundaryRecipe.test.tsx:68:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

68     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/GraphqlErrors.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:143:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

143     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:151:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

151     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/error-handling/LoadingErrorEmpty.test.tsx:159:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

159     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/hello/BindingAccountName.test.tsx:55:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

55     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ChildToParent.test.tsx:76:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

76     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ConditionalStatus.test.tsx:69:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

69     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/HelloWorld.test.tsx:28:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

28     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/LifecycleFetch.test.tsx:100:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

100     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ListOfAccounts.test.tsx:78:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

78     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/ParentToChild.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/hello/StateManagement.test.tsx:90:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

90     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/integration/DashboardAliasedQueries.test.tsx:89:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

89     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/integration/SearchableAccountList.test.tsx:168:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

168     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/integration/SearchableAccountList.test.tsx:179:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

179     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/CreateRecord.test.tsx:107:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

107     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/DeleteRecord.test.tsx:168:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

168     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/QueryMutationTogether.test.tsx:181:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

181     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/ServerErrorHandling.test.tsx:102:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

102     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/modify-data/UpdateRecord.test.tsx:160:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

160     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/AliasedMultiObjectQuery.test.tsx:60:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

60     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/FilteredList.test.tsx:121:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

121     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/ImperativeRefetch.test.tsx:86:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

86     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/ListOfRecords.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/PaginatedList.test.tsx:165:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

165     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/RelatedRecords.test.tsx:81:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

81     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/SingleRecord.test.tsx:75:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

75     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/read-data/SortedResults.test.tsx:98:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

98     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/LinkDemo.test.tsx:49:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

49     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/NavLinkDemo.test.tsx:52:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

52     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/routing/NestedRoutes.test.tsx:105:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

105     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/routing/RouteParameters.test.tsx:116:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

116     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/routing/UseNavigate.test.tsx:101:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

101     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/ApexRest.test.tsx:138:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

138     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/ConnectApi.test.tsx:136:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

136     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/DisplayCurrentUser.test.tsx:53:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

53     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/salesforce-apis/UiApiRest.test.tsx:148:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

148     expect(results).toHaveNoViolations();
                        ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardDSR.test.tsx:80:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

80     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardShadcn.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/AccountCardSLDS.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonDSR.test.tsx:60:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

60     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonShadcn.test.tsx:46:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

46     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonSLDS.test.tsx:21:21 - error TS2554: Expected 1 arguments, but got 2.

21       expect(match, `expected a button with class ${cls}`).toBeTruthy();
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/recipes/styling/ButtonSLDS.test.tsx:55:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

55     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsDSR.test.tsx:61:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

61     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsLucide.test.tsx:41:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

41     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsSLDS.test.tsx:50:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

50     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~


Found 49 errors.
```

> [@SCWells72](https://github.com/SCWells72) can you pull down the latest commit in [cw/fix-vitest-axe-windows](https://github.com/trailheadapps/multiframework-recipes/tree/cw/fix-vitest-axe-windows), remove node_modules, reinstall, and try building again? I updated vitest type declaration file to use @testing-library/jest-dom/vitest instead of @testing-library/jest-dom. This doesn't seem to matter for Mac/Linux but may for Windows.

Unfortunately that made it worse as it's now unable to resolve `toBeInTheDocument()` as well as `toHaveNoViolations()`:

```
npm run build

> react-recipes@1.59.0 build
> tsc -b && vite build

src/components/app/Layout.test.tsx:23:35 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.

23     expect(screen.getByText('A')).toBeInTheDocument();
                                     ~~~~~~~~~~~~~~~~~

src/components/app/Layout.test.tsx:25:39 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.

25     expect(screen.getByText('First')).toBeInTheDocument();
                                         ~~~~~~~~~~~~~~~~~

src/components/app/Layout.test.tsx:30:35 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.

30     expect(screen.getByText('C')).toBeInTheDocument();
                                     ~~~~~~~~~~~~~~~~~

...

src/recipes/styling/IconsSLDS.test.tsx:13:42 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.

13     expect(screen.getByText('Standard')).toBeInTheDocument();
                                            ~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsSLDS.test.tsx:14:40 - error TS2339: Property 'toBeInTheDocument' does not exist on type 'JestMatchers<HTMLElement>'.

14     expect(screen.getByText('Action')).toBeInTheDocument();
                                          ~~~~~~~~~~~~~~~~~

src/recipes/styling/IconsSLDS.test.tsx:50:21 - error TS2339: Property 'toHaveNoViolations' does not exist on type 'JestMatchers<AxeResults>'.

50     expect(results).toHaveNoViolations();
                       ~~~~~~~~~~~~~~~~~~


Found 327 errors.
```

--- a/.github/workflows/ci-pr.yml
+++ b/.github/workflows/ci-pr.yml
@@ -155,6 +155,31 @@ jobs:
           echo One of more Code Analyzer critical or high severity violations found
           exit 1
 
+  windows-build:
+    if: github.actor != 'trailheadapps-bot'
+    runs-on: windows-latest
+    steps:
+      - name: "Checkout source code"
+        uses: actions/checkout@v4
+
+      - name: "Setup Node"
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: npm
+          cache-dependency-path: force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json
+
+      - name: "Install React app dependencies"
+        run: npm ci --prefix force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Build"
+        run: npm run build
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Test"
+        run: npx vitest --run
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+
   scratch-org-test:
     runs-on: trailheadapps-Ubuntu
     if: github.actor != 'dependabot[bot]'

--- a/.github/workflows/ci-pr.yml
+++ b/.github/workflows/ci-pr.yml
@@ -157,28 +157,7 @@ jobs:
 
   windows-build:
     if: github.actor != 'trailheadapps-bot'
-    runs-on: windows-latest
-    steps:
-      - name: "Checkout source code"
-        uses: actions/checkout@v4
-
-      - name: "Setup Node"
-        uses: actions/setup-node@v4
-        with:
-          node-version: 22
-          cache: npm
-          cache-dependency-path: force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json
-
-      - name: "Install React app dependencies"
-        run: npm ci --prefix force-app/main/react-recipes/uiBundles/reactRecipes
-
-      - name: "Build"
-        run: npm run build
-        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
-
-      - name: "Test"
-        run: npx vitest --run
-        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+    uses: ./.github/workflows/windows-build.yml
 
   scratch-org-test:
     runs-on: trailheadapps-Ubuntu

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -138,6 +138,30 @@ jobs:
           echo One of more Code Analyzer critical or high severity violations found
           exit 1
 
+  windows-build:
+    runs-on: windows-latest
+    steps:
+      - name: "Checkout source code"
+        uses: actions/checkout@v4
+
+      - name: "Setup Node"
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: npm
+          cache-dependency-path: force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json
+
+      - name: "Install React app dependencies"
+        run: npm ci --prefix force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Build"
+        run: npm run build
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Test"
+        run: npx vitest --run
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+
   scratch-org-test:
     runs-on: trailheadapps-Ubuntu
     if: github.actor != 'dependabot[bot]'

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -139,28 +139,7 @@ jobs:
           exit 1
 
   windows-build:
-    runs-on: windows-latest
-    steps:
-      - name: "Checkout source code"
-        uses: actions/checkout@v4
-
-      - name: "Setup Node"
-        uses: actions/setup-node@v4
-        with:
-          node-version: 22
-          cache: npm
-          cache-dependency-path: force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json
-
-      - name: "Install React app dependencies"
-        run: npm ci --prefix force-app/main/react-recipes/uiBundles/reactRecipes
-
-      - name: "Build"
-        run: npm run build
-        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
-
-      - name: "Test"
-        run: npx vitest --run
-        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+    uses: ./.github/workflows/windows-build.yml
 
   scratch-org-test:
     runs-on: trailheadapps-Ubuntu

--- a/.github/workflows/windows-build.yml
+++ b/.github/workflows/windows-build.yml
@@ -1,0 +1,37 @@
+name: Windows Build
+
+on:
+  workflow_call:
+
+jobs:
+  windows-build:
+    runs-on: windows-latest
+    steps:
+      - name: "Checkout source code"
+        uses: actions/checkout@v4
+
+      - name: "Setup Node"
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "package.json"
+          cache: npm
+          cache-dependency-path: force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json
+
+      - name: "Restore React app node_modules cache"
+        id: cache-react-npm
+        uses: actions/cache@v4
+        with:
+          path: force-app/main/react-recipes/uiBundles/reactRecipes/node_modules
+          key: windows-react-npm-${{ hashFiles('force-app/main/react-recipes/uiBundles/reactRecipes/package-lock.json') }}
+
+      - name: "Install React app dependencies"
+        if: steps.cache-react-npm.outputs.cache-hit != 'true'
+        run: npm ci --prefix force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Build"
+        run: npm run build
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes
+
+      - name: "Test"
+        run: npm run test -- --run
+        working-directory: force-app/main/react-recipes/uiBundles/reactRecipes

--- a/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
+++ b/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
@@ -1,12 +1,4 @@
 /// <reference types="vitest" />
-/// <reference types="@testing-library/jest-dom" />
+/// <reference types="@testing-library/jest-dom/vitest" />
 /// <reference types="vitest/globals" />
-
-import type { AxeMatchers } from 'vitest-axe/matchers';
-
-declare module 'vitest' {
-  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
-  interface Assertion extends AxeMatchers {}
-  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
-  interface AsymmetricMatchersContaining extends AxeMatchers {}
-}
+/// <reference types="vitest-axe/extend-expect" />

--- a/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
+++ b/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
@@ -1,4 +1,4 @@
-/// <reference types="vitest" />
-/// <reference types="@testing-library/jest-dom" />
-/// <reference types="vitest/globals" />
-/// <reference types="vitest-axe/extend-expect" />
+import 'vitest';
+import 'vitest/globals';
+import '@testing-library/jest-dom';
+import 'vitest-axe/extend-expect';

--- a/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
+++ b/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
@@ -1,4 +1,4 @@
-/// <reference types="vitest" />
-/// <reference types="@testing-library/jest-dom/vitest" />
-/// <reference types="vitest/globals" />
-/// <reference types="vitest-axe/extend-expect" />
+import 'vitest';
+import 'vitest/globals';
+import '@testing-library/jest-dom';
+import 'vitest-axe/extend-expect';

--- a/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
+++ b/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
@@ -1,4 +1,12 @@
 /// <reference types="vitest" />
 /// <reference types="@testing-library/jest-dom" />
 /// <reference types="vitest/globals" />
-/// <reference types="vitest-axe/extend-expect" />
+
+import type { AxeMatchers } from 'vitest-axe/matchers';
+
+declare module 'vitest' {
+  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
+  interface Assertion extends AxeMatchers {}
+  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
+  interface AsymmetricMatchersContaining extends AxeMatchers {}
+}

--- a/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
+++ b/force-app/main/react-recipes/uiBundles/reactRecipes/vitest-env.d.ts
@@ -1,4 +1,4 @@
-import 'vitest';
-import 'vitest/globals';
-import '@testing-library/jest-dom';
-import 'vitest-axe/extend-expect';
+/// <reference types="vitest" />
+/// <reference types="@testing-library/jest-dom" />
+/// <reference types="vitest/globals" />
+/// <reference types="vitest-axe/extend-expect" />

